src/passes/externalArgumentModifier/dynamicArrayReturner.ts

```diff
@@ -0,0 +1,68 @@
+import assert = require('assert');
+
+import {
+  ArrayType,
+  DataLocation,
+  Expression,
+  FunctionDefinition,
+  getNodeType,
+  PointerType,
+  Return,
+  TupleExpression,
+  TypeNode,
+} from 'solc-typed-ast';
+
+import { AST } from '../../ast/ast';
+
+import { ASTMapper } from '../../ast/mapper';
+import { isExternallyVisible } from '../../utils/utils';
+export class DynArrayReturner extends ASTMapper {
+  /*
+  This pass looks for expressions that are returning dynamic arrays from external functions. Once those expressions are found it makes sure that
+  the correct StructDefinitions are created to hold the pointers of dynamic arrays.
+  */
+
+  visitFunctionDefinition(node: FunctionDefinition, ast: AST): void {
+    const body = node.vBody;
+
+    if (
+      isExternallyVisible(node) &&
+      body !== undefined &&
+      body.lastChild instanceof Return &&
```

Comment:
Returns don't always have to be the last child of the body. e.g. if there's an `if` then it'll have two returns inside it